tdrs-frontend/src/actions/reports.js

```diff
@@ -269,8 +270,20 @@ export const SET_SELECTED_YEAR = 'SET_SELECTED_YEAR'
 export const SET_SELECTED_QUARTER = 'SET_SELECTED_QUARTER'
 export const SET_FILE_TYPE = 'SET_FILE_TYPE'
 
-export const setStt = (stt) => (dispatch) => {
-  dispatch({ type: SET_SELECTED_STT, payload: { stt } })
+export const setStt = (stt) => async (dispatch) => {
+  var response = undefined
+  if (stt !== '') {
+    const URL = `${process.env.REACT_APP_BACKEND_URL}/stts/${stt}/num_sections`
```

Comment:
i still think there's an opportunity to eliminate this api call somehow. `SET_SELECTED_STT` isn't dispatched until the "Search" button is pressed on the datafile page - so not only does it have to reach out for the submitted datafiles/submission history, a second api call needs to be made just for the number of sections

`SET_STTS` is dispatched whenever the frontend retrieves the stt list
![image](https://github.com/user-attachments/assets/5a50247f-82c7-4992-bd18-0d27e8cb04f8)

if each of the stts in the list could have `num_sections`, i think we could avoid this extra api request

Reply:
i think that could be as simple as modifying the `STTSerializer` on the backend to have a new generated field

Reply:
I see what you're saying, I think that is a better idea! I will make that change!